app/src/main/java/org/fossasia/openevent/general/event/EventsFragment.kt

```diff
@@ -96,14 +96,13 @@ class EventsFragment : Fragment() {
             startActivity(intent)
         }
 
-        showNoInternetScreen(isNetworkConnected())
+        showNoInternetScreen(eventsViewModel.isNetworkConnected())
```

Comment:
The viewmodel should push the change to the observing fragment instead of you querying the viewmodel

Reply:
> The viewmodel should push the change to the observing fragment instead of you querying the viewmodel

Then which logic should I move in respective ViewModels?

Reply:
Can I call it like `showNoInternetScreen(searchViewModel.network.isNetworkConnected()) ` directly and make `val network: Network` insted of ` private val network: Network` in view model?

Reply:
Look at how we are communicating with LiveData in other places